File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
from __future__ import annotations

RESISTOR_COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors: list[str]) -> int:
    return RESISTOR_COLORS.index(colors[0]) * 10 + RESISTOR_COLORS.index(colors[1])

File: solutions/ruby/resistor-color-duo/1/resistor_color_duo.rb
# Return the value of a resistor based on the first two bands
module ResistorColorDuo
  COLORS = %w[black brown red orange yellow green blue violet grey white].each_with_index.to_h

  def self.value(bands) = (COLORS[bands[0]] * 10) + COLORS[bands[1]]
end
